### PR TITLE
Do not include <arch>_fpmath.h

### PR DESCRIPTION
**Pull request by ginggs**
These headers do not appear to be needed to build libopenspecfun1.

### CODE REVIEW

**Comment by ginggs** [comment]
While trying to test why compiling libopenspecfun1 for armhf appeared to work, I plugged some test values into __ieee754_rem_pio2(x,y), __ieee754_rem_pio2f(x,y) and __kernel_rem_pio2(x,y,e0,nx,prec) and found I was getting the exact same results on armhf and on amd64.  I then found that I could remove amd64_fpmath.h and i386_fpmath.h without affecting the results.

**Comment by ViralBShah** [comment]
IIRC, these get used to improve performance where applicable, but are not strictly necessary. I feel it is best to just leave them as is for now.

**Comment by ViralBShah** [comment]
A better long term solution is perhaps to just get rid of all this C code and have a julia implementation.

Cc @simonbyrne

**Comment by ginggs** [comment]
In the short term, I'm trying to establish whether this code runs on architectures besides i386 and amd64 (at least armhf).  As you saw in Debian bug 799036 I'm trying to convince the maintainer to allow openspecfun to at least attempt to build on all architectures, at the moment it is restricted (in the debian/control file) to build on i386 and amd64 only.

[1] https://bugs.debian.org/799036

**Comment by ViralBShah** [comment]
This does build on armhf just fine as is.

File: rem_pio2/fpmath.h [hunk 1]
```diff
@@ -29,15 +29,6 @@
 #ifndef _FPMATH_H_
 #define _FPMATH_H_
 
-// Currently assumes Intel platform
```
Comment (pao) [inline rem_pio2/fpmath.h]: I can see where this comment would be confusing though. Should be something like:

```
// Additional optimizations for i386 and x86-64
```

**Comment by ginggs** [comment]
On amd64 I've built two versions of libopenspecfun.so.1.0; one including amd64_fpmath.h and one with amd64_fpmath.h removed (as per this patch).  The files are binary identical except for the .note.gnu.build-id section.

**Comment by ViralBShah** [comment]
Ok, perhaps not needed then. Can you time them both for performance? If identical, we can excise these bits.

**Comment by ginggs** [comment]
I modified my test program attached to the Debian bug to run from 0 to 2 * pi * 10^6 in steps of pi / 1024.  __ieee754_rem_pio2() and __kernel_rem_pio2() tests took around 180 seconds each, and __ieee754_rem_pio2f() took around 16 seconds.

Taking the best scores from three runs of each version, the results were within 0.34% and 0.49% of each other, with "no <arch>_fpmath.h headers" being slightly quicker in the __ieee754_rem_pio2()
 and __kernel_rem_pio2() tests, and "with <arch>_fpmath.h headers" being slightly quicker in the __ieee754_rem_pio2f() test.

**Comment by ViralBShah** [comment]
Thank you.

**Comment by tkelman** [comment]
Did you test on 32 bit builds?

**Comment by ginggs** [comment]
No, I'll try i386 now.

**Comment by ginggs** [comment]
Both i386 builds are binary identical except for the .note.gnu.build-id section.

Timings are also very similar; __ieee754_rem_pio2() 336 seconds with headers, 335 seconds without headers, __ieee754_rem_pio2f() 43 seconds with headers, 45 seconds without headers, and __kernel_rem_pio2() 323 seconds with headers, 324 without headers.

**Comment by tkelman** [comment]
Great, thanks for checking that.

**Pull request merged**